### ISSUE TITLE
Cursor does not blink in integrated terminal

### ISSUE THREAD

**Issue by CameronBieganek** [microsoft/vscode#131446]
Issue Type: <b>Bug</b>

Does this issue occur when all extensions are disabled?: Yes

Steps to reproduce:

1. Add `"terminal.integrated.cursorBlinking": true` to `settings.json` file.
2. Start VS Code with `code --disable-extensions`.
3. Launch integrated terminal (e.g. with CMD-J on Mac).
4. Observe that the cursor in the integrated terminal is not blinking, even when the focus is on the integrated terminal.

VS Code version: Code 1.59.1 (3866c3553be8b268c8a7f8c0482c0c0177aa8bfa, 2021-08-19T11:53:52.479Z)
OS version: Darwin x64 18.7.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|16.00GB (8.02GB free)|
|Process Argv|--disable-extensions --crash-reporter-id fe789409-e4d3-492d-b5c7-952d2b0ca735|
|Screen Reader|no|
|VM|0%|
</details>Extensions disabled<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805cf:30301675
pythonvspyt200:30340761
vscextlang:30333561
binariesv615:30325510
pythonvssor306:30344512
bridge0708:30335490
vstre464:30350172
bridge0723:30353136
vsdyn420:30354109

```

</details>

**Comment by flaviocamilo** [comment]
This also happens on Linux.

Maybe something related to the updated version of Xterm.js, which resolved #106981?

**Comment by TarunavBA** [comment]
Dudes @CameronBieganek and @flaviocamilo you both can use the terminal, it has no issues. There is nothing wrong with the terminal if the cursor isn't blinking.

**Comment by CameronBieganek** [comment]
@TarunavBA On the contrary, this affects the usability of the terminal. Here's why:

The purpose of a blinking cursor is to visually indicate which pane/window within VS Code is currently active. Without a blinking cursor in the terminal, it is much harder to tell if the terminal is currently the active pane. The result is that I sometimes think I'm in the terminal and start typing a terminal command, but that text is instead inserted into a source file in the source editor, because it turns out that the source editor was the active pane rather than the integrated terminal. This is pretty annoying and a time waster.

**Comment by meganrogge** [comment]
Works on Windows, haven't tried it on mac or linux. Maybe related to https://github.com/xtermjs/xterm.js/commit/8b5a1331068a6bfb9d2face5c3f16c64fed8f8d7 ? @Tyriar

**Comment by TarunavBA** [comment]
If you are on Mac or Linux you can check the color of the minimize maximise and close buttons. In which window it is colored that window is active. 🤷‍♂️

**Comment by larseirikhansen** [comment]
This is an issue for me as well, and has been for a while. I have tested this on Mac and Linux (Mint 18.x and Zorin 16) with both code-insiders and the stable version. I have also tried turning all extensions off, but the problem persists. I have not

For me, the cursor does blink the first time opening vscode and the integrated terminal. Then, if I kill the terminal and create a new, it stops blinking. 

The only way to make it _temporarily_ start again is to either reload the window, or change any of the settings that changes how the terminal-cursor renders (`terminal.integrated.cursorWidth`, `terminal.integrated.cursorStyle`, `terminal.integrated.cursorBlinking`, etc.). 

Here is a gif I just made to show. 
![Untitled (4)](https://user-images.githubusercontent.com/71271458/132007283-79737489-5458-4891-93f1-cdfe56842e1d.gif)

**Comment by CameronBieganek** [comment]
> If you are on Mac or Linux you can check the color of the minimize maximise and close buttons. In which window it is colored that window is active.

No, that tells you which macOS window is active. What I was referring to is which _pane_ within the VS Code application is the active pane, and there is no colorization that indicates which pane is active. For example, here's a screenshot of VS Code with the integrated terminal open, and you can see that there is nothing to indicate whether the editor is active or the integrated terminal is active:

<img width="1440" alt="Screen Shot 2021-09-03 at 4 37 09 PM" src="https://user-images.githubusercontent.com/8310743/132067783-8d51292d-2f43-4119-b094-c19fe0b037a6.png">

**Comment by Tyriar** [comment]
I can't repro either, but looking at the code I'm guessing the issue is caused because we don't init the cursor settings in the terminal ctor:

https://github.com/microsoft/vscode/blob/c2955308e54a3d4cff62ac0956cb32f5048eb171/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L573-L596

Perhaps a config update events is firing after that on some setups which causes the cursor settings to apply.

**Comment by meganrogge** [comment]
reopening because of reports that this is still broken

**Comment by ryenus** [comment]
The problem is that when setting `terminal.integrated.cursorBlinking` to `true`, it **ONLY works with the currently opened terminal sessions**.

If the terminal pane is closed, by clicking "Close Panel" (`X`) button, then when reopening the same terminal sessions. the cursor still blinks.

![image](https://user-images.githubusercontent.com/610161/135176733-c448d08a-3f9d-4bd9-802e-070615741e9c.png)

But if I quit the terminal by terminating the shell, either via `exit` or <kbd> ctrl + d </kbd>, then the new integrated terminal sessions do not have blinking cursors even though the setting is still on.

---

As [noted in the other issue](https://github.com/microsoft/vscode/issues/133608#issuecomment-929013072):

### Steps
1. open the fresh insider build
2. open the integrated terminal, and focus it
3. notice the terminal cursor doesn't blink, this is expected since "terminal.integrated.cursorBlinking" defaults to false
4. open settings, turn on terminal cursor blinking
5. now focus the integrated terminal again, see that the terminal cursor starts blinking
6. run the `exit` command, or press <kbd>ctrl + d</kbd>, RATHER THAN closing the terminal panel
7. **again, open a new integrated terminal, here the terminal cursor DOES NOT blink, even though the setting is still on.**

And the problem is with the last step above, apparently things are different regarding the terminal is whether a new one or not.

I just tested this with the latest Insider build and the problem still exists:
> Version: 1.61.0-insider (Universal)
Commit: 214ac689ca9874a8a4be386f73d5cb60e15af25e
Date: 2021-09-28T05:14:06.244Z
Electron: 13.4.0
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0

**Comment by pogilvie** [comment]
At some point during the day I end up typing terminal commands into source code because of this.  I'm a command line guy and I live in vscode so it's causing a lot of pain here.

**Comment by meganrogge** [comment]
while we work to fix this, you can turn `terminal.integrated.gpuAcceleration` to `off` or `canvas` - it's working properly for those renderers.
